src/details/ArborX_DetailsBoundingVolumeHierarchyImpl.hpp

```diff
@@ -95,16 +93,18 @@ void BoundingVolumeHierarchyImpl<DeviceType>::queryDispatch(
   bool const use_deprecated_nearest_query_algorithm =
       which == NearestQueryAlgorithm::PriorityQueueBased_Deprecated;
 
-  auto const n_queries = queries.extent(0);
+  using Access = Traits::Access<Predicates, Traits::PredicatesTag>;
+  auto const n_queries = Access::size(predicates);
 
   Kokkos::Profiling::pushRegion("ArborX:BVH:sort_queries");
 
   auto const permute =
       Details::BatchedQueries<DeviceType>::sortQueriesAlongZOrderCurve(
-          bvh.bounds(), queries);
+          bvh.bounds(), predicates);
 
-  queries =
-      Details::BatchedQueries<DeviceType>::applyPermutation(permute, queries);
+  // FIXME  readability!  queries is a sorted copy of the predicates
+  auto queries = Details::BatchedQueries<DeviceType>::applyPermutation(
+      permute, predicates);
```

Comment:
So this doesn't impact performance?

Reply:
This does not change the algorithm and the choices that were previously done, which was to make a sorted copy of the queries/predicates instead of accessing with indirection.  I had also explored populating the results in permuted order and reverse the permutation at the end but that was slower.
We can try to not make a copy (i.e. drop `queries`) and do `Access::get(predicates, permute(i))` but I don't want to explore this in this PR.  Also I am pretty sure I had explored that option back then.